src/subscriptions/subscriptionBackfill.ts

```diff
@@ -129,15 +135,25 @@ export function makeBackfiller(jsonRpcSenders: JsonRpcSenders) {
     if (fromBlockInclusive >= toBlockExclusive) {
       return [];
     }
-    const batchParts: BatchPart[] = [];
+    let batchParts: BatchPart[] = [];
+    const headEvents: Array<Promise<BlockHead[]>> = [];
```

Comment:
(Nit) I find `headEvents` a bit of a confusing name. These are more like batches of head events, and promises of them at that. I might call this something like `headEventBatches` or `futureHeadBatches`.